### ISSUE TITLE
new preference for editor window size vs console window size

### ISSUE THREAD

**Issue by ottenm** [processing/processing#5119]
I'd like a way to set the height of the editor window compared with the height of the console window.   Could be in lines, pixels, percent, ... anything.  I just want the editor window to take up most of the pde, and only a few lines at the bottom for the black console window.   editor.window.height.default is close, but only controls the overall pde height (editor + console combined).

**Comment by dzaima** [comment]
You can move the console divider up and down, or does that not work for you?

**Comment by ottenm** [comment]
Yes, but that's what I'd like to have in preferences so I don't have to
manually move it down every time I open Processing, somewhere between 1 and
about 30 times per day.  Sorry I didn't know the name of it.  Would love to
set the "console divider" location in preferences, just like the other
window attributes in preferences.txt, most of which can also be changed
manually.

**Comment by benfry** [comment]
That's already stored in the preferences and should be working the way you describe it. 

Can you give us more details about your setup so we can figure out why it's not working for you?

**Comment by ottenm** [comment]
So sorry if I've missed it!

Not sure which preference it is.  These two look close.  I tried adding 10
to each of them independently, but couldn't detect any visible differences
when opening Processing....

console.lines=4
editor.divider.size=2

**Comment by benfry** [comment]
It should be saving the position for you between uses—it's not a preference that you should have to modify yourself.

**Comment by ottenm** [comment]
No luck.  I've moved that thing down a few hundred times and have never
seen it open (from scratch) down where I like it (maybe 10% off the bottom
of the window).  If I open Processing, and move it down, and open a second
Processing window, the second window does open as expected.  But if I close
out of everything the divider goes back up to about 25% of the overall
window when I reopen.  Same result if I open a sketch, or just open to a
blank window....

**Comment by tolkanabroski** [comment]
@ottenm 

A workaround could be:
 + delete old files C:\Users\[USERNAME]\AppData\Roaming\Processing
 (you can find the path to preferences.txt in Processing IDE under Files > Preferences) 
 + restart processing > new preferences.txt will be created
 + close Processing IDE
 + open preferences.txt with your fav. editor 
 + edit the settings
 + close everything 
 + resart processing

I'm on Windows and just edit a .txt file does nothing. Also the checkbox at Welcome Messege don't work with an old installation of Processing upgraded to a new verison. I always have to delete the old preferences.txt whait til a new one is created edit this one reopen everything etc.

**Comment by GKFX** [comment]
@benfry What preference is supposed to be automatically set? I see the same thing on my machine, even after deleting all of `~/.processing` and with the current build. Also, a diff on my `preferences.txt` before and after moving the console shows nothing except the instance server state changing, and I can't find where the preference is set in the code either (although that doesn't mean it's not there).

**Comment by benfry** [comment]
The location is stored as part of `EditorState`, but a quick glance shows that perhaps it's not getting saved to the prefs (anymore?) Hm…

**Comment by ottenm** [comment]
Can definitely give it a try.  Anyone know *which* preference it is?  It's
not getting updated in preferences.txt for me, but once I move the divider
and then open a second window (without closing Processing), the second one
does use the position I set in the first window....

**Comment by GKFX** [comment]
@ottenm So it seems that there is no preference being set automatically (no preference at all as far as I can see), but EditorState does get passed between windows which is why you see what you see.
